src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -912,6 +971,44 @@ public FormValidation doTestConnection(
             }
         }
 
+        @RequirePOST
+        @SuppressWarnings({"unused", "lgtm[jenkins/csrf]"
+        }) // used by jelly and already fixed jenkins security scan warning
+        public FormValidation doCheckSkipTlsVerify(@QueryParameter boolean skipTlsVerify) {
+            Jenkins.get().checkPermission(Jenkins.MANAGE);
+            try {
+                checkSkipTlsVerify(skipTlsVerify);
+            } catch (IllegalArgumentException ex) {
+                return FormValidation.error(ex, ex.getLocalizedMessage());
+            }
```

Comment:
**question:** I am guessing that your idea is to avoid code duplication, and that you require a method doing the actual check and returning an Exception for using in DataboundSetters and the like? 

Because if feels quite weird in these `doCheck` methods to actually rely on these exceptions to do the validation, especially later in other validations where you use `RuntimeException`.

What's the rationale for that design instead of using some other approach not requiring to catch these exceptions?

Reply:
Good catch with the `RuntimeException`, they should be `IllegalArgumentException` (autocpmplete changed it for an `IllegalStateException` 🤦 ).Changing it now.

Reply:
Mainly to avoid duplication. 
We could return a `FormValidation` object with an ok or the exception + message instead of throwing a exception, but then we would be checking in a `FormValidation` object in all the databoundsetters to launch the exception, which doesn't sound right.
We could also add another layer, having
* check* methods perform the logic and returns an error string | null
* assert* methods invoke check* and if return !null raise an exception
* doCheck* methods invoke check* and return a FormValidation with the contents.

The problem I see with this approach is that we might have other exceptions (like in [this](https://github.com/jenkinsci/kubernetes-plugin/pull/1590/files#diff-e862fe8545670a6000fc68dfe0444442a4d10bd88933533cd7668fb91eca4992R39-R42) test). 
In this test we're trying to check validity of something that is not a certificate (not an invalid certificate, directly not a certificate). This raises a `DecoderException` and we want to keep the stack trace (that points to the original class finding the problem in bouncy castle). If the check* methods were doing this check we would lose the original exception.